source/postcard-schema/src/schema/mod.rs

```diff
@@ -104,75 +90,78 @@ pub enum DataModelType {
     Unit,
 
-    /// The "unit struct" Serde Data Model Type
-    UnitStruct,
-
-    /// The "newtype struct" Serde Data Model Type
-    NewtypeStruct(&'static NamedType),
-
     /// The "Sequence" Serde Data Model Type
-    Seq(&'static NamedType),
+    Seq(&'static Self),
 
     /// The "Tuple" Serde Data Model Type
-    Tuple(&'static [&'static NamedType]),
-
-    /// The "Tuple Struct" Serde Data Model Type
-    TupleStruct(&'static [&'static NamedType]),
+    Tuple(&'static [&'static Self]),
 
     /// The "Map" Serde Data Model Type
     Map {
         /// The map "Key" type
-        key: &'static NamedType,
+        key: &'static Self,
         /// The map "Value" type
-        val: &'static NamedType,
+        val: &'static Self,
     },
 
-    /// The "Struct" Serde Data Model Type
-    Struct(&'static [&'static NamedValue]),
+    /// One of the struct Serde Data Model types
+    Struct {
+        /// The name of this struct
+        name: &'static str,
+        /// The data contained in this struct
+        data: Data,
+    },
 
     /// The "Enum" Serde Data Model Type (which contains any of the "Variant" types)
-    Enum(&'static [&'static NamedVariant]),
+    Enum {
+        /// The name of this struct
+        name: &'static str,
+        /// The variants contained in this enum
+        variants: &'static [&'static Variant],
+    },
 
-    /// A NamedType/OwnedNamedType
+    /// A [`DataModelType`]/[`OwnedDataModelType`](owned::OwnedDataModelType)
     Schema,
 }
 
-/// This is similar to [`DataModelType`], however it only contains the potential Data Model Types
-/// used as variants of an `enum`.
+/// The contents of a struct or enum variant.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize)]
-pub enum DataModelVariant {
-    /// The "unit variant" Serde Data Model Type
-    UnitVariant,
-    /// The "newtype variant" Serde Data Model Type
-    NewtypeVariant(&'static NamedType),
-    /// The "Tuple Variant" Serde Data Model Type
-    TupleVariant(&'static [&'static NamedType]),
-    /// The "Struct Variant" Serde Data Model Type
-    StructVariant(&'static [&'static NamedValue]),
+pub enum Data {
+    /// The "Unit Struct" or "Unit Variant" Serde Data Model Type
+    Unit,
+
+    /// The "Newtype Struct" or "Newtype Variant" Serde Data Model Type
+    Newtype(&'static DataModelType),
+
+    /// The "Tuple Struct" or "Tuple Variant" Serde Data Model Type
+    Tuple(&'static [&'static DataModelType]),
+
+    /// The "Struct" or "Struct Variant" Serde Data Model Type
+    Struct(&'static [&'static NamedField]),
 }
 
 /// This represents a named struct field.
 ///
 /// For example, in `struct Ex { a: u32 }` the field `a` would be reflected as
-/// `NamedValue { name: "a", ty: DataModelType::U32 }`.
+/// `NamedField { name: "a", ty: DataModelType::U32 }`.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize)]
-pub struct NamedValue {
-    /// The name of this value
+pub struct NamedField {
```

Comment:
I thought this name made a bit more sense, but I can revert it if we want to minimize breaking changes